### PR TITLE
TK3.4: add service args to get-http-show

### PR DESCRIPTION
**Pull request by frenchy64**
Related https://github.com/threatgrid/iroh/issues/3986

`ctia.properties/get-http-show` is at the root of a large function dependency chain, so there's lots of changes needed for it to support `get-in-config`.

<a name="qa">[§](#qa)</a> QA
============================

No QA is needed.

<a name="release-notes">[§](#release-notes)</a> Release Notes
=============================================================

```
intern: TK3.4: add service args to get-http-show
```

<a name="squashed-commits">[§](#squashed-commits)</a> Squashed Commits
======================================================================

### CODE REVIEW

File: src/ctia/domain/entities.clj [hunk 1]
```diff
@@ -60,34 +60,34 @@
...
 
-(defn with-long-id [entity]
-  (update entity :id short-id->long-id))
+(defn with-long-id [entity get-in-config]
+  (update entity :id short-id->long-id get-in-config))
 
-(defn page-with-long-id [m]
-  (update m :data #(map with-long-id %)))
-
-(def ->long-id (id/factory:short-id+type->long-id get-http-show))
```
Comment (frenchy64) [inline src/ctia/domain/entities.clj:75]: Unused.

File: src/ctia/entity/feed/schemas.clj [hunk 2]
```diff
@@ -81,7 +82,7 @@
     owner :- s/Str
     groups :- [s/Str]
     prev-object :- (s/maybe StoredFeed)]
-   (let [long-id (short-id->long-id id)
+   (let [long-id (short-id->long-id id p/get-in-global-properties)
```
Comment (frenchy64) [inline src/ctia/entity/feed/schemas.clj:85]: This function body will be wrapped in a function taking services. For now, using global properties.

File: src/ctia/schemas/graphql/resolvers.clj [hunk 3]
```diff
@@ -3,6 +3,7 @@
             [ctia.domain.entities
              :refer
              [page-with-long-id un-store un-store-page with-long-id]]
+            [ctia.properties :as p]
```
Comment (frenchy64) [inline src/ctia/schemas/graphql/resolvers.clj:6]: GraphQL resolvers will be changed soon to accept a map of services. For now, using global properties.

File: src/ctia/domain/entities.clj [hunk 4]
```diff
@@ -60,34 +60,34 @@
...
+                        %)))
 
 (defn long-id->entity-type [id-str]
   (:type (id/long-id->id id-str)))
 
-(defn short-id->entity-type [id-str]
+(defn short-id->entity-type [id-str get-in-config]
   (when-let [short-id (id/short-id->long-id
                        id-str
-                       (get-http-show))]
```
Comment (frenchy64) [inline src/ctia/domain/entities.clj:83]: This was a bug, should have been `get-http-show`, not `(get-http-show)`.

**Review by gbuisson — APPROVED** [review]
LGTM!

**Review by ereteog — APPROVED** [review]
lgtm

**Pull request merged**